openwisp_controller/geo/tests/test_selenium.py

```diff
@@ -46,3 +59,54 @@ def _fill_device_form(self):
         )
         self.find_element(by=By.CLASS_NAME, value='select2-results__option').click()
         super()._fill_device_form()
+
+
+@tag('selenium_tests')
+class TestDeviceAdminUnsavedChangesGeoSelenium(
```

Comment:
A more generic naming would welcome adding more tests for readonly users to this class:

```suggestion
class TestDeviceAdminReadonly(
```